File: LargeTruck.java
public class LargeTruck extends Truck {
    public static final int MAX_VOLUME = 4000;
    public static final int MAX_WEIGHT = 8000;

    public LargeTruck() {
        super();
    }

    @Override
    public void addPack(Pack pack) {
        if (totalVolume() + pack.getVolume() <= MAX_VOLUME &&
                totalWeight() + pack.getWeight() <= MAX_WEIGHT) {
            packages.add(pack);
        }
    }

    @Override
    public int totalVolume() {
        int total = 0;
        for (Pack pack : packages) {
            total += pack.getVolume();
        }
        return total;
    }

    @Override
    public int totalWeight() {
        int total = 0;
        for (Pack pack : packages) {
            total += pack.getWeight();
        }
        return total;
    }
}

File: C#/PackageDeliverySystem/LargeTruck.cs
namespace PackageDeliverySystem
{
    class LargeTruck : Truck
    {
        public static int LARGE_MAX_VOLUME = 4000;
        public static int LARGE_MAX_WEIGHT = 8000;

        public LargeTruck() : base() { }

        public override void addPack(Pack pack)
        {
            if (totalVolume() + pack.Volume <= LARGE_MAX_VOLUME &&
                totalWeight() + pack.Weight <= LARGE_MAX_WEIGHT)
            {
                base.addPack(pack);
            }
        }

        public override int getMaxWeight()
        {
            return LARGE_MAX_WEIGHT; 
        }

        public override int getMaxVolume()
        {
            return LARGE_MAX_VOLUME;
        }
    }
}
